src/core/obj.js

```diff
@@ -32,578 +32,600 @@ function fetchDestination(dest) {
+        } else {
+          currentIndex = 1;
         }
+      }
 
-        switch (style) {
-          case 'D':
-            currentLabel = currentIndex;
-            break;
-          case 'R':
-          case 'r':
-            currentLabel = toRomanNumerals(currentIndex, style === 'r');
-            break;
-          case 'A':
-          case 'a':
-            var LIMIT = 26; // Use only the characters A--Z, or a--z.
-            var A_UPPER_CASE = 0x41, A_LOWER_CASE = 0x61;
-
-            var baseCharCode = (style === 'a' ? A_LOWER_CASE : A_UPPER_CASE);
-            var letterIndex = currentIndex - 1;
-            var character = String.fromCharCode(baseCharCode +
+      switch (style) {
+        case 'D':
+          currentLabel = currentIndex;
+          break;
+        case 'R':
+        case 'r':
+          currentLabel = toRomanNumerals(currentIndex, style === 'r');
+          break;
+        case 'A':
+        case 'a':
+          const LIMIT = 26; // Use only the characters A-Z, or a-z.
+          const A_UPPER_CASE = 0x41, A_LOWER_CASE = 0x61;
+
+          const baseCharCode = (style === 'a' ? A_LOWER_CASE : A_UPPER_CASE);
+          const letterIndex = currentIndex - 1;
+          const character = String.fromCharCode(baseCharCode +
                                                 (letterIndex % LIMIT));
-            var charBuf = [];
-            for (var j = 0, jj = (letterIndex / LIMIT) | 0; j <= jj; j++) {
-              charBuf.push(character);
-            }
-            currentLabel = charBuf.join('');
-            break;
-          default:
-            if (style) {
-              throw new FormatError(
-                `Invalid style "${style}" in PageLabel dictionary.`);
-            }
-            currentLabel = '';
-        }
+          const charBuf = [];
+          for (let j = 0, jj = (letterIndex / LIMIT) | 0; j <= jj; j++) {
+            charBuf.push(character);
+          }
+          currentLabel = charBuf.join('');
+          break;
+        default:
+          if (style) {
+            throw new FormatError(
+              `Invalid style "${style}" in PageLabel dictionary.`);
+          }
+          currentLabel = '';
+      }
+
+      pageLabels[i] = prefix + currentLabel;
+      currentIndex++;
+    }
+    return pageLabels;
+  }
+
+  get pageMode() {
+    const obj = this.catDict.get('PageMode');
+    let pageMode = 'UseNone'; // Default value.
 
-        pageLabels[i] = prefix + currentLabel;
-        currentIndex++;
+    if (isName(obj)) {
+      switch (obj.name) {
+        case 'UseNone':
+        case 'UseOutlines':
+        case 'UseThumbs':
+        case 'FullScreen':
+        case 'UseOC':
+        case 'UseAttachments':
+          pageMode = obj.name;
       }
-      return pageLabels;
-    },
+    }
+    return shadow(this, 'pageMode', pageMode);
+  }
+
+  get attachments() {
+    let attachments = null, nameTreeRef;
+    const obj = this.catDict.get('Names');
+    if (obj) {
```

Comment:
This is another potential for clean-up (again in a separate commit).
It looks like this code was basically copy-pasted from an older version of the `destinations` getter, but it could probably be changed to e.g.
```javascript
 get attachments() {
   const obj = this.catDict.get('Names');
   let attachments = null;

   if (obj && obj.has('EmbeddedFiles')) {
     const nameTree = new NameTree(obj.getRaw('EmbeddedFiles'), this.xref);
     const names = nameTree.getAll();
     for (let name in names) {
       const fs = new FileSpec(names[name], this.xref);
       if (!attachments) {
         attachments = Object.create(null);
       }
       attachments[stringToPDFString(name)] = fs.serializable;
     }
   }
   return shadow(this, 'attachments', attachments);
}
```